v2/tools/asoctl/internal/importing/arm_resource_importer_factory_test.go

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"github.com/Azure/azure-service-operator/v2/api"
+	. "github.com/onsi/gomega"
+	"testing"
+)
+
+func Test_ARMResourceImporterFactory_GroupKindFromArmId(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name          string
+		armId         string
+		expectedGroup string
+		expectedKind  string
+	}{
+		{
+			name:          "resource group",
+			armId:         "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/aso-rg",
+			expectedGroup: "resources.azure.com",
+			expectedKind:  "ResourceGroup",
+		},
+		{
+			name:          "storage account",
+			armId:         "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/aso-rg/providers/Microsoft.Storage/storageAccounts/asostorage",
+			expectedGroup: "storage.azure.com",
+			expectedKind:  "StorageAccount",
+		},
+		{
+			name:          "cosmos account",
+			armId:         "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/aso-rg/providers/Microsoft.DocumentDB/databaseAccounts/aso-cosmos",
+			expectedGroup: "documentdb.azure.com",
+			expectedKind:  "DatabaseAccount",
+		},
+		{
+			name:          "VMSS Scale Set	",
+			armId:         "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/aso-rg/providers/Microsoft.Compute/virtualMachineScaleSets/aso-scaleset",
+			expectedGroup: "compute.azure.com",
+			expectedKind:  "VirtualMachineScaleSet",
+		},
+		{
```

Comment:
We should also support extension resources (which are weird). The best example of one which we support today in ASO is a roleAssignment.

Reply:
Extensions are now suported.